apps/api/src/app/http-exception.filter.ts

```diff
@@ -0,0 +1,23 @@
+import { ExceptionFilter, Catch, ArgumentsHost, HttpException, Logger } from '@nestjs/common'
+import { Request, Response } from 'express'
+
+@Catch()
+export class HttpExceptionFilter implements ExceptionFilter {
```

Comment:
So if I understood correctly this is an error handling middleware?

Reply:
Precisely! The framework has a nice built-in global error handler already, but it was omitting errors from Zod schemas. I had no idea what was failing so I wrote another one myself, just to `console.log` the error

In the end, it helped me to identify that I was typing `.optional()` instead of `.nullable()`